fluent-fallback/src/bundles.rs

```
@@ -0,0 +1,425 @@
+use crate::{
+    cache::{AsyncCache, Cache},
+    env::LocalesProvider,
+    errors::LocalizationError,
+    generator::{BundleGenerator, BundleIterator, BundleStream},
+    types::{L10nAttribute, L10nKey, L10nMessage},
+};
+use fluent_bundle::{FluentArgs, FluentBundle, FluentError};
+use std::borrow::Cow;
+
+pub enum BundlesInner<G>
+where
+    G: BundleGenerator,
+{
+    Iter(Cache<G::Iter, G::Resource>),
+    Stream(AsyncCache<G::Stream, G::Resource>),
+}
+
+pub struct Bundles<G>(BundlesInner<G>)
+where
+    G: BundleGenerator;
+
+impl<G> Bundles<G>
+where
+    G: BundleGenerator,
+    G::Iter: BundleIterator,
+{
+    pub fn prefetch_sync(&self) {
+        match &self.0 {
+            BundlesInner::Iter(iter) => iter.prefetch(),
+            BundlesInner::Stream(_) => todo!(),
```

Feels a bit weird to use `todo!()` for this right? This should probably return an error or do something else.